src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -873,11 +882,42 @@ void SolverInterfaceImpl::setMeshQuad(
     PRECICE_CHECK(mesh->isValidEdgeID(secondEdgeID), errorInvalidEdgeID(secondEdgeID));
     PRECICE_CHECK(mesh->isValidEdgeID(thirdEdgeID), errorInvalidEdgeID(thirdEdgeID));
     PRECICE_CHECK(mesh->isValidEdgeID(fourthEdgeID), errorInvalidEdgeID(fourthEdgeID));
-    mesh::Edge &e0 = mesh->edges()[firstEdgeID];
-    mesh::Edge &e1 = mesh->edges()[secondEdgeID];
-    mesh::Edge &e2 = mesh->edges()[thirdEdgeID];
-    mesh::Edge &e3 = mesh->edges()[fourthEdgeID];
-    mesh->createQuad(e0, e1, e2, e3);
+
+    PRECICE_CHECK(utils::unique_elements(utils::make_array(firstEdgeID, secondEdgeID, thirdEdgeID, fourthEdgeID)),
+                  "The four edge ID's are not unique. Please check that the edges that form the quad are correct.");
+
+    auto chain = mesh::asChain(utils::make_array(
+        &mesh->edges()[firstEdgeID], &mesh->edges()[secondEdgeID],
+        &mesh->edges()[thirdEdgeID], &mesh->edges()[fourthEdgeID]));
+    PRECICE_CHECK(chain.connected, "The four edges are not connect. Please check that the edges that form the quad are correct.");
+
+    auto coords = mesh::coordsFor(chain.vertices);
+    PRECICE_CHECK(utils::unique_elements(coords),
+                  "The four vertices that form the quad are not unique. The resulting shape may be a point, line or triangle."
+                  "Please check that the adapter sends the four unique vertices that form the quad, or that the mesh on the interface "
+                  "is composed of quads. A mix of triangles and quads are not supported.");
+
+    auto convexity = math::geometry::isConvexQuad(coords);
+    PRECICE_CHECK(convexity.convex, "The given quad is not convex. "
+                                    "Please check that the adapter send the four correct vertices or that the interface is composed of quads. "
+                                    "A mix of triangles and quads are not supported.");
```

Comment:
Why again the "A mix of triangles and quads are not supported"?

Reply:
I didn't understand it either. Removing (twice).